muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/records/body/application/records.tsx

```diff
@@ -45,52 +40,43 @@ function getEvaluationRequestIfAvailable(props: RecordsProps, workspace: Workspa
     assesmentState = workspace.studentAssessmentState.state;
     assesmentDate = workspace.studentAssessmentState.date;
   } else if (workspace.studentActivity && workspace.studentActivity.assessmentState){
-    assesmentState = workspace.studentActivity.assessmentState.state;
-    assesmentDate = workspace.studentActivity.assessmentState.date;
+    assesmentState = workspace.studentAssessmentState.state;
+    assesmentDate = workspace.studentAssessmentState.date;
   }
-  
+
   if (assesmentState === "pending" || assesmentState === "pending_pass" || assesmentState === "pending_fail"){
-    return <div className="application-list__header-secondary">
-      <span>{props.i18n.text.get("plugin.records.workspace.pending",props.i18n.time.format(assesmentDate))}</span>
-      <span title={props.i18n.text.get("plugin.records.workspace.pending",props.i18n.time.format(assesmentDate))}
-        className="application-list__indicator-badge application-list__indicator-badge--evaluation-request icon-assessment-pending"></span>
-    </div>
+    return <span title={props.i18n.text.get("plugin.records.workspace.pending",props.i18n.time.format(assesmentDate))} className="application-list__indicator-badge application-list__indicator-badge--evaluation-request icon-assessment-pending"></span>
   }
-
   return null;
 }
 
 function getTransferCreditValue(props: RecordsProps, transferCredit: TransferCreditType){
-  return <div className="application-list__header-secondary">
-    <span>{props.i18n.text.get("plugin.records.transferCreditsDate", props.i18n.time.format(transferCredit.date))}</span>
-    <span title={props.i18n.text.get("plugin.records.transferCreditsDate", props.i18n.time.format(transferCredit.date)) +
-      getShortenGradeExtension(transferCredit.grade)} className={`application-list__indicator-badge application-list__indicator-badge-course ${transferCredit.passed ? "state-PASSED" : "state-FAILED"}`}>
-      {shortenGrade(transferCredit.grade)}
+  let gradeId = [
+    transferCredit.gradingScaleIdentifier,
+    transferCredit.gradeIdentifier].join('-');
+  let grade = props.records.grades[gradeId];
```

Comment:
props.records.grades seems undefined. UI value for grade can be gotten via transferCredit.grade and UI value for grading scale  if needed - via transferCredit.gradingScale. In the code below, whether the grade is passing or failed can be gotten via transferCredit.passing.